api/src/main/java/com/velocitypowered/api/command/CommandManager.java

```diff
@@ -35,10 +35,21 @@
 
   /**
    * Attempts to execute a command from the specified {@code cmdLine}.
+   * CommandExecuteEvent will not called
    *
    * @param source the command's source
    * @param cmdLine the command to run
    * @return true if the command was found and executed, false if it was not
    */
   boolean execute(CommandSource source, String cmdLine);
+
+  /**
+   * Attempts to execute a command from the specified {@code cmdLine}.
+   *
+   * @param source the command's source
+   * @param cmdLine the command to run
+   * @param callEvent will CommandExecuteEvent called or not
+   * @return true if the command was found and executed, false if it was not
+   */
+  boolean execute(CommandSource source, String cmdLine, boolean callEvent);
```

Comment:
This is the biggest problem I have with this. I am not a big fan of exposing a method like this. Could we do a bit better?

On an additional note, I think we should probably deprecate `execute` and friends and just outright replace them with `*async` variants. We'll clean up the API mess later on.

Reply:
I think its not problem, because execute will be called from event/executor thread pool in 99% cases. I think we should not messy this with async stuff.